core/broadleaf-framework/src/main/java/org/broadleafcommerce/core/offer/service/processor/AbstractBaseProcessor.java

```diff
@@ -497,17 +501,17 @@ protected List<Offer> removeOutOfDateOffers(List<Offer> offers){
     }
 
     protected Calendar dateToCalendar(Date date, TimeZone offerTimeZone) {
-        DateTimeZone offerDateTimeZone = DateTimeZone.forTimeZone(offerTimeZone);
-        LocalDateTime offerDateTime = new LocalDateTime(date, offerDateTimeZone);
+        ZonedDateTime offerDateTimeZone = ZonedDateTime.ofInstant(Instant.from((TemporalAccessor) date), ZoneId.of(String.valueOf(offerTimeZone)));
+        LocalDateTime offerDateTime = LocalDateTime.ofInstant(Instant.from((TemporalAccessor) date), ZoneId.from(offerDateTimeZone));
```

Comment:
maybe we need just LocalDateTime offerDateTime = LocalDateTime.ofInstant(Instant.from((TemporalAccessor) date), ZoneId.of(String.valueOf(offerTimeZone))); ?
so we don't need offerDateTimeZone ?

Reply:
There are test failures which I think is because of casting here:
```
[ERROR] testApplyAllFulfillmentGroupOffersWithOrderItemOffers(org.broadleafcommerce.core.offer.service.processor.FulfillmentGroupOfferProcessorTest)  Time elapsed: 0.034 s  <<< ERROR!
java.lang.ClassCastException: java.util.Date cannot be cast to java.time.temporal.TemporalAccessor
	at org.broadleafcommerce.core.offer.service.processor.FulfillmentGroupOfferProcessorTest.testApplyAllFulfillmentGroupOffersWithOrderItemOffers(FulfillmentGroupOfferProcessorTest.java:275)
```